### PR TITLE
Speed up spd_execute_command_with_list_reply

### PR DESCRIPTION
**Pull request by novalis**
spd_execute_command_with_list_reply was accidentally quadratic: it would start by searching the string for one \r\n, then it would search for two, then for three...  I have 15,000 voices (I have no idea why; it seems to be the default Debian config), so this was quite slow.

For me, this improves the performance of spd_list_synthesis_voices from 1.5s to 0.5s.

### CODE REVIEW

**Comment by sthibaul** [comment]
Thanks for your patch, it looks promising :)

I'd however rather not copy/paste the implementation of `get_param_str`.

It seems to me like everything in `get_param_str` after the for loop could be replaced by a call to your function, thus factorizing everything that can be?

File: src/api/c/libspeechd.c [hunk 1]
```diff
@@ -542,7 +543,7 @@ SPDConnection *spd_open2(const char *client_name, const char *connection_name,
 	if (!connection->stream)
 		SPD_FATAL("Can't create a stream for socket, fdopen() failed.");
 	/* Switch to line buffering mode */
-	ret = setvbuf(connection->stream, NULL, _IONBF, SPD_REPLY_BUF_SIZE);
+	ret = setvbuf(connection->stream, NULL, _IOLBF, SPD_REPLY_BUF_SIZE);
```
Comment (sthibaul) [inline src/api/c/libspeechd.c:546]: Please do not mix with other fixes :)

**Review by sthibaul — CHANGES_REQUESTED** [review]

File: src/api/c/libspeechd.c (hunk 1 above)
Comment (novalis) [reply src/api/c/libspeechd.c]: oops

File: src/api/c/libspeechd.c [hunk 2]
```diff
@@ -2020,17 +2008,44 @@ static char *get_param_str(char *reply, int num, int *err)
 
 	pos_begin = pos + 4;
 	pos_end = strstr(pos_begin, "\r\n");
-	if (pos_end == NULL) {
+	if (*pos_end == 0) {
```
Comment (sthibaul) [inline src/api/c/libspeechd.c:2011]: Noticing this just now: there shouldn't be a `*`, it's really the pointer that needs to be tested. You can leave the code as it is.

**Review by sthibaul — APPROVED** [review]

**Comment by sthibaul** [comment]
Thanks!

**Pull request merged**